odoo/addons/base/migrations/12.0.1.3/pre-migration.py

```diff
@@ -170,6 +170,43 @@ def fill_ir_attachment_res_model_name(cr):
         )
 
 
+def fix_double_membership(cr):
+    # avoid error raised by new function '_check_one_user_type'
+
+    # assuming that group_public < group_portal < group_user
+    # this script keept the highest group, if a user belong to many
+    # groups
+    confs = [
+        ("group_public", "group_portal"),
+        ("group_public", "group_user"),
+        ("group_portal", "group_user"),
+    ]
+    for conf in confs:
+        group_to_remove = conf[0]
+        group_to_keep = conf[1]
+        openupgrade.logged_query(
+            cr, """
+                DELETE FROM res_groups_users_rel
+                WHERE
+                gid = (
+                    SELECT res_id
+                    FROM ir_model_data
+                    WHERE module = 'base' AND name = %s
+                )
+                AND uid IN (
+                    SELECT uid FROM res_groups_users_rel WHERE gid IN (
+                        SELECT res_id
+                        FROM ir_model_data
+                        WHERE module = 'base'
```

Comment:
```suggestion
                        WHERE module = 'base' AND model = 'res.groups'
```

Reply:
I don't get it. full xml id (module + name) is unique by design.